packages/suir-component-mapper/src/files/field-array.js

```diff
@@ -0,0 +1,269 @@
+import React, { useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { useFormApi, FieldArray } from '@data-driven-forms/react-form-renderer';
+
+import { Button, Icon, Header, ButtonGroup, ButtonContent } from 'semantic-ui-react';
+
+import { useFieldApi } from '@data-driven-forms/react-form-renderer';
+
+import { createUseStyles } from 'react-jss';
+import clsx from 'clsx';
+import FormField from '../common/form-field';
+
+const useStyles = createUseStyles({
+  buttonGroup: {
+    display: 'flex',
+    justifyContent: 'flex-end'
+  },
+  arrayItem: {
+    marginBottom: '1em'
+  },
+  arrayItems: {
+    marginTop: '1em'
+  },
+
+  error: {
+    color: '#9f3a38 !important'
+  },
+  noItems: {
+    paddingBottom: '1em'
+  },
+  noMargin: {
+    margin: '0 !important'
+  },
+  arrayHeader: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center'
+  }
+});
+
+const ArrayItem = ({ fields, fieldIndex, name, remove, length, minItems, removeLabel }) => {
+  const { renderForm } = useFormApi();
+  const classes = useStyles();
+
+  const editedFields = fields.map((field, index) => {
+    const computedName = field.name ? `${name}.${field.name}` : name;
+    return { ...field, name: computedName, key: `${computedName}-${index}` };
+  });
+
+  return (
+    <div className={classes.arrayItem}>
+      <div>{renderForm([editedFields])}</div>
+      <div>
+        <Button
+          type="button"
+          icon="remove"
+          content={removeLabel}
+          basic
+          color="red"
+          onClick={() => remove(fieldIndex)}
+          disabled={length <= minItems}
+        />
+      </div>
+    </div>
+  );
+};
+
+ArrayItem.propTypes = {
+  name: PropTypes.string,
+  fieldIndex: PropTypes.number.isRequired,
+  fields: PropTypes.arrayOf(PropTypes.object),
+  remove: PropTypes.func.isRequired,
+  length: PropTypes.number,
+  minItems: PropTypes.number,
+  removeLabel: PropTypes.node.isRequired
+};
+
+const defaultButtonLabels = {
+  add: 'ADD',
+  remove: 'REMOVE'
+};
+
+const initialState = {
+  index: 0,
+  history: []
+};
+
+export const reducer = (state, { type, action }) => {
```

Comment:
note for the future: we could make this common as it is being used on two places now